src/hssm/plotting/posterior_predictive.py

```diff
@@ -0,0 +1,199 @@
+        plot will be produced, by default None.
+    col_wrap : optional
+        “Wrap” the column variable at this width, so that the column facets span
+        multiple rows. Incompatible with a row facet., by default None.
+    palette : optional
+        Colors to use for the different levels of the hue variable. Should be something
+        that can be interpreted by color_palette(), or a dictionary mapping hue levels
+        to matplotlib colors., by default None.
+    binwidth : optional
+        A number or a pair of numbers. Width of each bin. There are two other binning
+        options: `bins` and `binrange`. Please see the documentation for the
+        [`sns.histplot` function]
+        (https://seaborn.pydata.org/generated/seaborn.histplot.html) for how to use
+        them. in combination to control the bins. This argument overrides bins but can
+        be used with `binrange.`
+    grid_kwargs : optional
+        Additional keyword arguments are passed to the [`sns.FacetGrid` constructor]
+        (https://seaborn.pydata.org/generated/seaborn.FacetGrid.html#seaborn.FacetGrid.__init__)
+        when any of row or col is provided. When producing a single plot, these
+        arguments are ignored.
+    kwargs : optional
+        Additional keyword arguments passed to the [sns.`histplot` function]
+        (https://seaborn.pydata.org/generated/seaborn.histplot.html).
+
+    Returns
+    -------
+    mpl.axes.Axes | sns.FacetGrid
+        The matplotlib axis or seaborn FacetGrid object containing the plot.
+    """
+    # First, determine whether posterior predictive samples are available
+    # If not, we need to sample from the posterior
+    if idata is None:
+        if model.traces is None:
+            raise ValueError(
+                "No InferenceData object provided. Please provide an InferenceData "
+                + "object or sample the model first using model.sample()."
+            )
+        idata = model.traces
+
+    if data is None:
+        data = model.data
+
+    extra_dims = [e for e in [row, col] if e is not None] or None
+
+    if "posterior_predictive" not in idata:
+        _logger.info(
+            "No posterior predictive samples found. Generating posterior predictive "
+            + "samples using the provided InferenceData object and the original data. "
+            + "This will modify the provided InferenceData object, or if not provided, "
+            + "the traces object stored inside the model."
+        )
+        model.sample_posterior_predictive(
+            idata=idata, data=data, inplace=True, n_samples=n_samples
+        )
+        plotting_df = _get_plotting_df(
+            idata, data, extra_dims=extra_dims, n_samples=None
+        )
+    else:
+        plotting_df = _get_plotting_df(
+            idata, data, extra_dims=extra_dims, n_samples=n_samples
+        )
+
+    # Flip the rt values if necessary
+    if np.any(plotting_df["rt"] == 0):
```

Comment:
this should happen only if `n_choices == 2`. 
if you apply this on other models you turn all non-zero responses to `1` ?

Reply:
Right now I am assuming `n_choices == 2` since we are not supporting more than 2 choices yet. We can revisit this once we do support more than 2 responses

Reply:
ok for now then